lib/msal-core/src/BrowserCrypto.ts

```diff
@@ -0,0 +1,134 @@
+        if (isIE11()) {
+            return new Promise((resolve, reject) => {
+                const signOperation = window['msCrypto'].subtle.sign(this._algorithmOptions, key, data);
+                signOperation.addEventListener('complete', (e: { target: { result: ArrayBuffer | PromiseLike<ArrayBuffer>; }; }) => {
+                    resolve(e.target.result);
+                });
+                signOperation.addEventListener('error', (error: any) => {
+                    reject(error);
+                });
+            });
+        }
+
+        return window.crypto.subtle.sign(this._algorithmOptions, key, data);
+    }
+
+    async verify(key: CryptoKey, signature: ArrayBuffer, data: ArrayBuffer): Promise<boolean> {
+        if (isIE11()) {
+            return new Promise((resolve, reject) => {
+                const verifySignatureOperation = window['msCrypto'].subtle.verify(this._algorithmOptions, key, signature, data);
+                verifySignatureOperation.addEventListener('complete', (e: { target: { result: boolean | PromiseLike<boolean>; }; }) => {
+                    resolve(e.target.result);
+                });
+                verifySignatureOperation.addEventListener('error', (error: any) => {
+                    reject(error);
+                });
+            });
+        }
+
+        return window.crypto.subtle.verify(this._algorithmOptions, key, signature, data);
+    }
+
+    async generateKey(extractable: boolean, usages: Array<string>): Promise<CryptoKeyPair> {
+        if (isIE11()) {
+            return new Promise((resolve, reject) => {
+                const generateKeyOperation = window['msCrypto'].subtle.generateKey(this._algorithmOptions, extractable, usages);
+                generateKeyOperation.addEventListener('complete', (e: { target: { result: CryptoKeyPair | PromiseLike<CryptoKeyPair>; }; }) => {
+                    resolve(e.target.result);
+                });
+                generateKeyOperation.addEventListener('error', (error: any) => {
+                    reject(error);
+                });
+            });
+        }
+
+        return window.crypto.subtle.generateKey(this._algorithmOptions, extractable, usages);
+    }
+
+    async exportKey(key: CryptoKey, format: KeyFormat): Promise<JsonWebKey> {
+        if (isIE11()) {
+            return new Promise((resolve, reject) => {
+                const exportKeyOperation = window['msCrypto'].subtle.exportKey(format, key);
+                exportKeyOperation.addEventListener('complete', (e: { target: { result: ArrayBuffer; }; }) => {
+                    const resultBuffer: ArrayBuffer = e.target.result;
+                    const resultString = getStringFromArrayBuffer(resultBuffer)
+                        .replace(/\r/g, '')
+                        .replace(/\n/g, '')
+                        .replace(/\t/g, '')
+                        .split(' ').join('')
+                        .replace("\u0000", '');
```

Comment:
can this be a single regex? seems like there is something you need to know about the result to format it, so if it changes, this string manipulation may break?

Reply:
maybe factor out the string manipulation into a format method you can test in isolation

Reply:
It could be a single regex, although I like that this is a little easier to comprehend. Given this is to normalize the response from IE11, I don't expect this to change, but agree about testing.